src/lib/optimizer/strategy/abstract_rule.hpp

```diff
@@ -50,15 +50,6 @@ class AbstractRule {
    * To traverse LQPs, use the visit_lqp function. Do not call this function recursively.
```

Comment:
Maybe we should add an explanation to this warning?

Suggestion:

> Do not call this function recursively. The reason for this can be found in diamond LQPs: When using "trivial" recursion, we would go down both on the left and the right side of the diamond. On both sides, we would reach the bottom of the diamon. From there, we would look at each node twice. visit_lqp prevents this by tracking which nodes have already been visited and avoiding visiting a node twice.

Reply:
Great! I added it.

Reply:
*diamond (if you haven't caught that typo anyway)

Reply:
caught it